includes/Settings.php

```diff
@@ -239,5 +262,22 @@ public function register() {
 				],
 			]
 		);
+
+		add_action( 'add_option_' . self::SETTING_NAME_ARCHIVE, [ $this, 'update_archive_setting' ] );
+		add_action( 'update_option_' . self::SETTING_NAME_ARCHIVE, [ $this, 'update_archive_setting' ] );
+	}
+
+
+	/**
+	 * Clear rewrite rules on update on setting.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return void
+	 */
+	public function update_archive_setting() {
```

Comment:
Can you confirm that the rewrite rule flushing works as expected?

In my testing it didn't seem to work. When I chose "Disabled" in the dropdown the archive page still worked, until I manually went to the Permalink settings to flush rewrite rules.

I think the issue here is that this calls `flush_rewrite_rules()`, but the story post type object still has `has_archive` set to the old value.

So we'd need to basically do something like this (pseudo code):

```php
global $wp_post_types;

$wp_post_types['web-story']->has_archive = 'default' === get_option( Settings::SETTING_NAME_ARCHIVE, 'default' );

flush_rewrite_rules( false );
```

Would be a reason to move this `'default' === get_option(...)` logic to the `Post_Type` trait.

Reply:
Been removed to another class and refactored.